includes/wf_crm_webform_postprocess.inc

```diff
@@ -1442,6 +1450,33 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
     }
   }
 
+  /**
+   * Handle adding/updating tags for entities (cases, activity)
+   *
+   * @param $entityType
+   * @param $entityId
+   * @param $params
+   */
+  function handleEntityTags($entityType, $entityId, $params) {
+    $tagValues = array();
+    foreach ($params as $param => $val) {
+      if ($val && strpos($param, 'tag_') === 0) {
+        $val = preg_replace('/[^A-Za-z0-9\-]/', ',', $val);
+        $val = array_filter(explode(',', $val), 'strlen');
+        foreach ($val as $tagValue) {
+          $tagValues[$tagValue] = $tagValue;
+        }
+      }
+    }
+    $allTags = civicrm_api3('EntityTag', 'getoptions', [
+      'field' => "tag_id",
+    ])['values'];
+    $allTags = array_keys($allTags);
+    $removeTags = array_diff($allTags, $tagValues);
+    $removeTags = array_combine($removeTags, $removeTags);
+    $this->addOrRemoveMultivaluedData('tag', $entityType, $entityId, $tagValues, $removeTags);
+  }
```

Comment:
@vinuvarshith Can you respond to coleman here please?

Reply:
@colemanw
What I am trying to do here is to support adding and untagging. With 'addOrRemoveMultivaluedData' we pass $removeTags param which should have the tags to be removed.
**What I tried previously:**
So if we try to remove a tag from webform by unchecking the checkbox, that value is not recorded and so is not handled. Here I am getting all tag ids, calculating a diff between that tags to be added and all tags which I consider as tags to be removed.
**Updated logic:**
Now I have updated this code to get the values from $this->crmValues. Is this ok? Should I also get the checked tag values from $this->crmValues too instead of params?

Reply:
Thanks for pointing this out. I have checked and fixed 'handleEntityTags' based on 'saveGroupsAndTags'. Tested in a clean setup and this works fine for me.
@colemanw